### PR TITLE
instrumentation telemetry event app-extended-heartbeat update

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
The `configuration` entry in app-extended-heartbeat used to contain exactly the same information as app-started, not taking into account that configuration values can change after app startup. This PR updates `configWithOrigin` whenever `updateConfig` is called, so that app-extended-heartbeat can send up-to-date configuration values.
This PR also contains unit tests for `updateConfig`

### Motivation
@CarlesDD asked about it

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: packages/dd-trace/test/telemetry/index.spec.js [hunk 1]
```diff
@@ -718,6 +718,71 @@ describe('Telemetry retry', () => {
...
+  }
+  const sendDataStub = sinon.stub()
+
+  const { updateConfig } = proxyquire('../../src/telemetry', {
+    './send-data': {
+      sendData: sendDataStub
+    }
+  })
+
+  it('should set configWithOrigin on initial call', () => {
```
Comment (rochdev) [inline packages/dd-trace/test/telemetry/index.spec.js:738]: This should describe the behaviour instead of the implementation.

File: packages/dd-trace/src/telemetry/index.js [hunk 2]
```diff
@@ -304,33 +304,45 @@ function updateConfig (changes, config) {
...
   }
+
+  function isNotModified (entry) {
+    return !names.includes(entry.name)
+  }
+
   if (!configWithOrigin.length) {
     configWithOrigin = configuration
   } else {
+    configWithOrigin = configWithOrigin.filter(isNotModified)
```
Comment (rochdev) [inline packages/dd-trace/src/telemetry/index.js:339]: I don't think this is supposed to be possible? Only changes should be propagated to begin with.

File: packages/dd-trace/src/telemetry/index.js [hunk 3]
```diff
@@ -304,33 +304,45 @@ function updateConfig (changes, config) {
...
 
-  const names = {
+  const nameMapping = {
     sampleRate: 'DD_TRACE_SAMPLE_RATE',
     logInjection: 'DD_LOG_INJECTION',
     headerTags: 'DD_TRACE_HEADER_TAGS',
     tags: 'DD_TAGS'
   }
 
+  const namesNeedFormatting = new Set(['DD_TAGS', 'peerServiceMapping', 'tags'])
```
Comment (rochdev) [inline packages/dd-trace/src/telemetry/index.js:314]: Why is there multiple keys for the same value?

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/telemetry/index.js [hunk 4]
```diff
@@ -304,33 +304,46 @@ function updateConfig (changes, config) {
...
     configWithOrigin = configuration
   } else {
+    // update configWithOrigin to contain up-to-date full list of config values for app-extended-heartbeat
+    configWithOrigin = configWithOrigin.filter(isNotModified)
+    configWithOrigin = configWithOrigin.concat(configuration)
     const { reqType, payload } = createPayload('app-client-configuration-change', { configuration })
     sendData(config, application, host, reqType, payload, updateRetryData)
   }
+
+  return configWithOrigin // for testing using sinon
```
Comment (rochdev) [inline packages/dd-trace/src/telemetry/index.js:346]: Why is this needed? It looks like the other tests are testing the telemetry actually being sent. Could this do the same? It would also result in tests with clearer descriptions.

**Review by Qard — APPROVED** [review]

**Review by rochdev — APPROVED** [review]
